storage/client/src/test/java/com/azure/storage/queue/QueueClientTests.java

```diff
@@ -0,0 +1,651 @@
+import static org.junit.Assert.fail;
+
+public class QueueClientTests extends QueueClientTestsBase {
+    private QueueClient client;
+
+    @Override
+    protected void beforeTest() {
+        queueName = getQueueName();
+
+        if (interceptorManager.isPlaybackMode()) {
+            client = setupClient((connectionString, endpoint) -> QueueClient.builder()
+                .connectionString(connectionString)
+                .endpoint(endpoint)
+                .queueName(queueName)
+                .httpClient(interceptorManager.getPlaybackClient())
+                .httpLogDetailLevel(HttpLogDetailLevel.BODY_AND_HEADERS)
+                .build());
+        } else {
+            client = setupClient((connectionString, endpoint) -> QueueClient.builder()
+                .connectionString(connectionString)
+                .endpoint(endpoint)
+                .queueName(queueName)
+                .httpClient(HttpClient.createDefault().wiretap(true))
+                .httpLogDetailLevel(HttpLogDetailLevel.BODY_AND_HEADERS)
+                .addPolicy(interceptorManager.getRecordPolicy())
+                .build());
+        }
+    }
+
+    @Override
+    protected void afterTest() {
+        try {
+            client.clearMessages();
+            client.delete();
+        } catch (StorageErrorException ex) {
+            // Queue already delete, that's what we wanted anyways.
+        }
+    }
+
+    @Override
+    public void createWithSharedKey() {
+
+    }
+
+    @Override
+    public void createWithSASToken() {
+        // Need to find a way to get SAS tokens from the storage account
+    }
+
+    @Override
+    public void createWithMetadata() {
+        Map<String, String> metadata = new HashMap<>();
+        metadata.put("metadata1", "value1");
+        metadata.put("metadata2", "value2");
+
+        TestHelpers.assertResponseStatusCode(client.create(metadata), 201);
+
+        QueueProperties properties = client.getProperties().value();
+        assertEquals(metadata, properties.metadata());
+    }
+
+    @Override
+    public void createTwiceSameMetadata() {
+        Map<String, String> metadata = new HashMap<>();
+        metadata.put("metadata1", "value1");
+        metadata.put("metadata2", "value2");
+
+        TestHelpers.assertResponseStatusCode(client.create(metadata), 201);
+        TestHelpers.assertResponseStatusCode(client.create(metadata), 204);
+    }
+
+    @Override
+    public void createTwiceDifferentMetadata() {
+        Map<String, String> metadata = new HashMap<>();
+        metadata.put("metadata1", "value1");
+        metadata.put("metadata2", "value2");
+
+        TestHelpers.assertResponseStatusCode(client.create(), 201);
+
+        try {
+            client.create(metadata);
+            fail("Creating a queue twice with different metadata values should throw an exception.");
+        } catch (Exception exception) {
+            TestHelpers.assertExceptionStatusCode(exception, 409);
```

Comment:
Just a thought (not a review) - I wonder what values we achieve by testing server side behavior like this. Do we want our build to break if server side changes this behavior, say, one day they just allow updating the metadata?

Reply:
I'd say that its useful for testing that we are doing correct error mapping/don't add any client side checks that we should allow the service to handle.